portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +27,33 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+    private List<String> messages;
+
+  @Override
+  public void init() {
+    messages = new ArrayList<>();
```

Comment:
You can initialize the list with the values already:
new ArrayList<>(Arrays.asList("a", "b", "c"));
Or alternatively use messages.addAll(Arrays.asList(..));

Generally speaking if you notice you're repeating the same method several times in a bulk, there's a use case for an aggregation of some sort and there's probably a way to do so:)